lib/src/main/java/com/knuddels/jtokkit/TokenEncoder.java

```diff
@@ -205,10 +205,8 @@ private int encode(ByteArrayWrapper payload) {
     }
 
     int encode(ByteArrayWrapper piece, int start, int end) {
-        if (end > piece.length()) {
+        if (end > piece.length() || end - start == piece.length()) {
```

Comment:
we've already checked the case when `start == 0` and `end == piece.length()` - i.e. the whole piece needs to be encoded letter-by-letter